File: 2021/2021W32/PreppindataW32.py
import pandas as pd
import numpy as np

sales = pd.read_csv("F:\\Data\\PD 2021 Wk 32 Input - Data.csv",
                    parse_dates=['Date','Date of Flight'], 
                    date_parser=lambda x: pd.to_datetime(x, format='%d/%m/%Y'))

sales['Flight'] = sales['Departure'] + ' to ' + sales['Destination'] 
sales['Days'] = (sales['Date of Flight'] - sales['Date']).dt.days
sales['Days Category'] = sales['Days'].apply(lambda x: '7 days of more until the flight' if x>=7 else 'less than 7 days until the flight')
sales = sales[['Flight', 'Class', 'Ticket Sales', 'Days Category']].copy()

final = np.round(sales.pivot_table(index=['Flight', 'Class'], columns='Days Category',
                                   values='Ticket Sales',
                                   aggfunc={'Ticket Sales':[('Sales', np.sum), ('Avg. daily sales', np.mean)]})).reset_index()
final.columns = [(c[0]+' '+c[1]).strip() for c in final.columns]
final = final[['Flight','Class',
               'Avg. daily sales 7 days of more until the flight',
               'Avg. daily sales less than 7 days until the flight',
               'Sales less than 7 days until the flight',
               'Sales 7 days of more until the flight']]

File: 2021/2021W32/preppindataW32.R
library(readr)
library(dplyr)
library(tidyr)
library(lubridate)

final <- read_csv("F:/Data/PD 2021 Wk 32 Input - Data.csv",
                  col_types = cols(Date = col_date(format = "%d/%m/%Y"), 
                                   `Date of Flight` = col_date(format = "%d/%m/%Y"))) %>%
  mutate('Flight' = paste(`Departure`,`Destination`, sep=' to '),
         'Days Category' = if_else(time_length(`Date of Flight`-`Date`, 'day')<7,
                                   'less than 7 days until the flight', '7 days of more until the flight'),
         'Sales' = `Ticket Sales`,
         'Avg. daily sales' = `Ticket Sales`) %>%
  select(c('Flight', 'Class', 'Sales', 'Avg. daily sales','Days Category')) %>%
  pivot_wider(id_cols=c('Flight', 'Class'), names_from='Days Category', names_sep = ' ',
              values_from=c('Sales','Avg. daily sales'),
              values_fn = list('Avg. daily sales' = mean, 'Sales' = sum)) %>%
  mutate(across(where(is.numeric), round, 0)) %>%
  select(c('Flight','Class',
           'Avg. daily sales 7 days of more until the flight',
           'Avg. daily sales less than 7 days until the flight',
           'Sales less than 7 days until the flight',
           'Sales 7 days of more until the flight'))

View(final)
